system/include/emscripten/bind.h

```diff
@@ -428,40 +449,10 @@ struct GetReturnValuePolicy<ReturnType, T, Rest...> {
 };
 
 template<typename... Policies>
-struct isAsync;
-
-template<typename... Rest>
-struct isAsync<async, Rest...> {
-    static constexpr bool value = true;
-};
-
-template<typename T, typename... Rest>
-struct isAsync<T, Rest...> {
-    static constexpr bool value = isAsync<Rest...>::value;
-};
-
-template<>
-struct isAsync<> {
-    static constexpr bool value = false;
-};
+using isAsync = disjunction<std::is_same<async, Policies>...>;
 
 template<typename... Policies>
-struct isNonnullReturn;
-
-template<typename... Rest>
-struct isNonnullReturn<nonnull<ret_val>, Rest...> {
-    static constexpr bool value = true;
-};
-
-template<typename T, typename... Rest>
-struct isNonnullReturn<T, Rest...> {
-    static constexpr bool value = isNonnullReturn<Rest...>::value;
-};
-
-template<>
-struct isNonnullReturn<> {
-    static constexpr bool value = false;
-};
+using isNonnullReturn = disjunction<std::is_same<nonnull<ret_val>, Policies>...>;
```

Comment:
Nice!

What are `async` and `non_null` in the above lines by the way?   Are they special values defines somewhere, or just some kind of placeholder?

Reply:
Policy args are basically options you can pass into function bindings. e.g.

```c++
   function('sleep', &sleep, async());
   function('constructor', &constructor, nonnull<ret_val>());
 ```